### PR TITLE
added —-build option to artifacts download

### PR DESCRIPTION
**Pull request by mgoodings**
Added option to specify the build id for artifact downloads.

### CODE REVIEW

**Comment by lox** [comment]
Awesome, thanks @mgoodings!

File: hooks/pre-command [hunk 1]
```diff
@@ -26,6 +26,12 @@ done < <(env | sort)
 
 
 if [ "${#paths[@]}" -gt 0 ]; then
-  echo "--- Downloading artifacts"
-  buildkite-agent artifact download "$(IFS=, ; echo "${paths[*]}")" "${BUILDKITE_PLUGIN_ARTIFACTS_WORKDIR}"
+  build=${BUILDKITE_PLUGIN_ARTIFACTS_BUILD:-}
```
Comment (lox) [inline hooks/pre-command]: Generally the best way to handle this conditional arguments stuff is with an `args` array, like in https://github.com/buildkite-plugins/docker-buildkite-plugin/blob/master/hooks/command#L12

File: hooks/pre-command [hunk 2]
```diff
@@ -26,6 +26,12 @@ done < <(env | sort)
...
+  build_message=$([[ -n "$build" ]] && echo " from build: $build" || echo "")
+
+  echo "--- Downloading artifacts$build_message"
```
Comment (lox) [inline hooks/pre-command]: I'd tend towards an if/else with duplicated echo lines for the two header types, it's more verbose, but the code will read better.

**Comment by mgoodings** [comment]
Any chance on getting this merged?

**Comment by toolmantim** [comment]
Sorry for the delay! This looks good. Will merge and put out a release now.

**Comment by toolmantim** [comment]
I've added the options to the readme, and released v1.1.0. Thanks for the contribution!

**Pull request merged**